indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3863,6 +3891,17 @@ public void setPartitionIdsForTests(
     partitionIds.addAll(partitionIdsForTests);
   }
 
+  @VisibleForTesting
+  public Map<String, Task> createActiveTaskMap(List<Task> tasks)
+  {
+    Map<String, Task> taskMap = new HashMap<>();
```

Comment:
Couple of suggestions here:
- do the `taskStorage.getActiveTasks()` call inside this method. Otherwise, this method is just converting the list to a map.
- return an immutable map from here
- perhaps rename it to `getActiveTasksMap` rather than `create...`
- keep this method private as there is hardly anything to test here. It will be tested via the other methods anyway.

Reply:
Done